AmplifyPlugins/API/Tests/APIHostApp/GraphQLAPIStressTests/GraphQLAPIStressTests.swift

```diff
@@ -0,0 +1,355 @@
+            }
+        }
+
+        await waitForExpectations([postCreateExpectation], timeout: TestCommonConstants.networkTimeout)
+        
+        for index in 0..<concurrencyLimit {
+            Task {
+                var post = await postOffice.posts[index]
+                let deletedPost = try await deletePost(post: post)
+                XCTAssertNotNil(deletedPost)
+                XCTAssertEqual(post.id, deletedPost.id)
+                XCTAssertEqual(post.title, deletedPost.title)
+                await postDeleteExpectation.fulfill()
+            }
+        }
+        
+        await waitForExpectations([postDeleteExpectation], timeout: TestCommonConstants.networkTimeout)
+    }
+    
+    /// - Given: APIPlugin configured with valid configuration, schema and 50 posts saved
+    /// - When: I query for 50 posts simultaneously
+    /// - Then: Operation should succeed
+    func testMultipleQueryByID() async throws {
+        let postCreateExpectation = asyncExpectation(description: "Post was created successfully",
+                                                     expectedFulfillmentCount: concurrencyLimit)
+        let postQueryExpectation = asyncExpectation(description: "Post was deleted successfully",
+                                                     expectedFulfillmentCount: concurrencyLimit)
+        let postOffice = PostOffice()
+        
+        for index in 0..<concurrencyLimit {
+            Task {
+                let id = UUID().uuidString
+                let title = "title" + String(index)
+                let post = try await createPost(id: id, title: title)
+                XCTAssertNotNil(post)
+                XCTAssertEqual(id, post?.id)
+                XCTAssertEqual(title, post?.title)
+                await postOffice.append(post: post!)
+                await postCreateExpectation.fulfill()
+            }
+        }
+
+        await waitForExpectations([postCreateExpectation], timeout: TestCommonConstants.networkTimeout)
+        
+        for index in 0..<concurrencyLimit {
+            Task {
+                var post = await postOffice.posts[index]
+                let graphQLResponse = try await Amplify.API.query(request: .get(Post.self, byId: post.id))
+                guard case .success(let data) = graphQLResponse else {
+                    XCTFail("Missing successful response")
+                    return
+                }
+                guard let queriedPost = data else {
+                    XCTFail("Missing post from query")
+                    return
+                }
+                XCTAssertNotNil(queriedPost)
+                XCTAssertEqual(post.id, queriedPost.id)
+                XCTAssertEqual(post.title, queriedPost.title)
+                await postQueryExpectation.fulfill()
+            }
+        }
+        
+        await waitForExpectations([postQueryExpectation], timeout: TestCommonConstants.networkTimeout)
+    }
+    
+    actor PostOffice {
```

Comment:
just curious, why is this called `PostOffice`?

Reply:
Renamed